src/pages/docs/registration-authorities/acme-for-vault.mdx

```diff
@@ -0,0 +1,266 @@
+Success! You are now authenticated. The token information displayed below
+is already stored in the token helper. You do NOT need to run "vault login"
+again. Future Vault requests will automatically use this token.
+
+Key                  Value
+---                  -----
+token                hvs.VGBPeFkQtkGp5sDdHBug0Bge
+token_accessor       kGo8ZgkOlrw9Fa1neEjDH52m
+token_duration       ∞
+token_renewable      false
+token_policies       ["root"]
+identity_policies    []
+policies             ["root"]
+```
+
+### 2. Create your Root CA:
+
+Now run: 
+
+```bash
+ROOT_CA_COMMON_NAME="Vault Root CA"
+vault secrets enable pki
+vault secrets tune -max-lease-ttl=87600h pki
+vault write pki/root/generate/internal \
+	common_name="$ROOT_CA_COMMON_NAME" \
+	ttl=87600h
+vault read -format=json pki/cert/$(vault list -format=json pki/certs | jq -r '.[0]') | jq -r .data.certificate > root_ca.crt
+vault write pki/config/urls \
+	issuing_certificates="$VAULT_ADDR/v1/pki/ca" \
+	crl_distribution_points="$VAULT_ADDR/v1/pki/crl"
+```
+
+Your root CA is created,
+and your root CA certificate PEM is saved in `root_ca.crt`.
+
+### 3. Create your Intermediate CA:
+
+Now let's create and sign the intermediate CA using the root CA.
+
+```bash
+INTERMEDIATE_CA_COMMON_NAME="Vault Intermediate CA"
+vault secrets enable -path=pki_int1 pki
+vault secrets tune -max-lease-ttl=43800h pki_int1
+vault write -format=json pki_int1/intermediate/generate/internal \
+	common_name="$INTERMEDIATE_CA_COMMON_NAME" \
+	ttl=43800h  | jq -r .data.csr > int1.csr
+vault write -format=json pki/root/sign-intermediate \
+	csr=@int1.csr \
+	format=pem_bundle \
+	ttl=43800h | jq -r .data.certificate > int1.crt
+cat int1.crt root_ca.crt > intermediate_ca.crt
+vault write pki_int1/intermediate/set-signed \
+	certificate=@intermediate_ca.crt
+vault write pki_int1/config/urls \
+	issuing_certificates="$VAULT_ADDR/v1/pki_int1/ca" \
+	crl_distribution_points="$VAULT_ADDR/v1/pki_int1/crl"
+```
+
+Your intermediate CA is created,
+and your root and intermediate CA PEM bundle is saved in `intermediate_ca.crt`.
+
+### 3. Create Vault roles and policy
+
+Now let's create Vault roles that your RA server will use to sign certificates.
+To create these, run:
+
+```bash
+vault write pki_int1/roles/ec-role key_type=ec allow_any_name=true enforce_hostnames=false max_ttl=72h
+vault write pki_int1/roles/rsa-role key_type=rsa allow_any_name=true enforce_hostnames=false max_ttl=72h
+vault write pki_int1/roles/ed25519-role key_type=ed25519 allow_any_name=true enforce_hostnames=false max_ttl=72h
+```
+
+You'll also need policies for each role. Run:
+
+```bash
+echo 'path "pki_int1/sign/ec-role" {
+    capabilities = ["create", "update"]
+}
+path "pki_int1/sign/rsa-role" {
+    capabilities = ["create", "update"]
+}
+path "pki_int1/sign/ed25519-role" {
+    capabilities = ["create", "update"]
+}
+path "pki_int/revoke" {
+    capabilities = ["create", "update"]
+}' > policy.hcl
+vault policy write smallstep policy.hcl
```

Comment:
According to @brotherdust, this was too permissive.

Reply:
I'm sorry I haven't been able to address this yet. I should clarify: it's not that it's too permissive, it's that the policy enables the use of basically all the supported algorithms. The documentation should specify which cipher families to use in which context. For instance, it might not be a good idea to allow RSA on an EC CA, or ed25519 on a CA that will be issuing TLS certs for web services (compatibility reasons). 

Perhaps that is out of the scope of this document; in which case, it would be good to provide a link to reference material that would allow the user to make informed choices about which cipher families to allow via policy. 

Does that make sense? Maybe I'm just overthinking this. Sorry if that's the case!

Reply:
Makes sense, but I think the default configuration should allow signing any kind of certificate no matter the key it is using. We can add links to vault docs for more information. 

The same happens with role configurations, if you want to restrict some specific hostnames, you can also do this on vault, but the default role allows everything. You will also be able to manage this in the upcoming policy in `step-ca` instead of doing it on Vault.

Reply:
I agree and I will add a sentence to clarify this.